SocialCareCaseViewerApi/V1/Boundary/Requests/ListAllocationsRequest.cs

```diff
@@ -11,6 +11,9 @@ public class ListAllocationsRequest
 
         [FromQuery(Name = "worker_id")]
         public long WorkerId { get; set; }
+
+        [FromQuery(Name = "worker_email")]
```

Comment:
The way the boundary object validation was initially setup means that providing only `worker_email` will result in 400 error, so minor tweaks are required for the request validator.

Reply:
I've done my best here, used Fluent validations with two custom validators (couldn't find ones that matched) for "At least one of these properties must be set" and "only one of these properties can be set".

Reply:
Might be nice to make the namespace nullable, then instead of assuming defaults (0) as no property passed we'd actually be checking for null/!null values

Reply:
> Might be nice to make the namespace nullable, then instead of assuming defaults (0) as no property passed we'd actually be checking for null/!null values

Do you have an example of this in action?

Reply:
```
#nullable enable
namespace SocialCareCaseViewerApi.V1.Boundary.Requests
{
    public class ListAllocationsRequest
    {
        [FromQuery(Name = "mosaic_id")]
        public long? MosaicId { get; set; }

        [FromQuery(Name = "worker_id")]
        public long? WorkerId { get; set; }

        [FromQuery(Name = "worker_email")]
        public string? WorkerEmail { get; set; }
    }

    public class ListAllocationsRequestValidator : AbstractValidator<ListAllocationsRequest>
    {
        public ListAllocationsRequestValidator()
        {
            RuleFor(x => x)
                .Must(OneIsSet)
                .WithMessage("Please provide either mosaic_id, worker_id or worker_email");

            RuleFor(x => x)
                .Must(OnlyOneIsSet)
                .WithMessage("Please provide only one of mosaic_id, worker_id or worker_email");

            When(x => !String.IsNullOrEmpty(x.WorkerEmail), () =>
            {
                RuleFor(x => x.WorkerEmail)
                    .EmailAddress()
                    .WithMessage("Please provide a valid email address for worker_email");
            });
        }

        private bool OnlyOneIsSet(ListAllocationsRequest request)
        {
            return new List<int>
            {
                request.MosaicId == null ? 0 : 1,
                request.WorkerId == null ? 0 : 1,
                String.IsNullOrEmpty(request.WorkerEmail) ? 0 : 1,
            }.Sum() <= 1;
        }

        private bool OneIsSet(ListAllocationsRequest request)
        {
            return !(request.MosaicId == null && request.WorkerId == null && String.IsNullOrEmpty(request.WorkerEmail));
        }
    }
```

Reply:
I've found being explicit about what can and can't be null has resulted in a substantial decrease in server side errors

Reply:
Ok, cool, I'll get that change in, will save the refactoring for another task because of time pressure.

Reply:
are we happy to merge as is for now?